wardrobe/tradable_kitties/src/tests.rs

```diff
@@ -0,0 +1,864 @@
+        &[TradableKittyData::default_tradable_kitty().into()],
+        &[],
+        &[],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::NumberOfInputOutputMismatch)
+    );
+}
+
+#[test]
+fn delist_kitty_from_sale_with_wrong_output_type_ampoung_valid_output_fails() {
+    let output1 = TradableKittyData::default_kitty();
+    let mut output2 = TradableKittyData::default_kitty();
+    output2.dna = KittyDNA(H256::from_slice(b"superkalifragislisticexpialadoca"));
+    let mut input1 = TradableKittyData::default_tradable_kitty();
+    let mut input2 = TradableKittyData::default_tradable_kitty();
+    input1.kitty_basic_data = output1.clone();
+    input2.kitty_basic_data = output2.clone();
+
+    let result = TradableKittyConstraintChecker::<0>::DelistKittyFromSale.check(
+        &[input1.into(), input2.into()],
+        &[],
+        &[output1.into(), Bogus.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::BadlyTyped));
+}
+
+#[test]
+fn delist_from_sale_with_wrong_input_type_fails() {
+    let result = TradableKittyConstraintChecker::<0>::DelistKittyFromSale.check(
+        &[Bogus.into()],
+        &[],
+        &[TradableKittyData::default_kitty().into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::BadlyTyped));
+}
+
+#[test]
+fn delist_from_sale_with_duplicate_input_fails() {
+    let output1 = TradableKittyData::default_kitty();
+    let mut output2 = TradableKittyData::default_kitty();
+    output2.dna = KittyDNA(H256::from_slice(b"superkalifragislisticexpialadocz"));
+    let mut input1 = TradableKittyData::default_tradable_kitty();
+    let mut input2 = TradableKittyData::default_tradable_kitty();
+    input1.kitty_basic_data = output1.clone();
+    input2.kitty_basic_data = output2.clone();
+    let result = TradableKittyConstraintChecker::<0>::DelistKittyFromSale.check(
+        &[input1.clone().into(), input1.into()],
+        &[],
+        &[output1.into(), output2.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::DuplicateTradableKittyFound)
+    );
+}
+
+#[test]
+fn delist_from_sale_with_duplicate_output_fails() {
+    let output1 = TradableKittyData::default_kitty();
+    let mut output2 = TradableKittyData::default_kitty();
+    output2.dna = KittyDNA(H256::from_slice(b"superkalifragislisticexpialadocz"));
+    let mut input1 = TradableKittyData::default_tradable_kitty();
+    let mut input2 = TradableKittyData::default_tradable_kitty();
+    input1.kitty_basic_data = output1.clone();
+    input2.kitty_basic_data = output2.clone();
+    let result = TradableKittyConstraintChecker::<0>::DelistKittyFromSale.check(
+        &[input1.clone().into(), input2.into()],
+        &[],
+        &[output1.clone().into(), output1.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::DuplicateKittyFound));
+}
+
+// From below update tradable kitty name test cases starts
```

Comment:
If you keep the update name function, then you need a test to make sure that no basic properties of the kitty change during renames.

Reply:
I added test cases for ensuring no basic properties of the kitty change during renames.